beeai/issue_fetcher/jira_issue_fetcher.py

```diff
@@ -0,0 +1,381 @@
+                                        schema = NoActionData.model_validate_json(item)
+                                        issue_key = schema.jira_issue
+                                    elif queue_name in ["error_list"]:
+                                        schema = ErrorData.model_validate_json(item)
+                                        issue_key = schema.jira_issue
+                                except ValueError:
+                                    # Fallback to task parsing for these queues if direct parsing fails
+                                    task = Task.model_validate_json(item)
+                                    if task.metadata and "issue" in task.metadata:
+                                        issue_key = task.metadata["issue"]
+                            
+                            if issue_key:
+                                existing_keys.add(issue_key)
+                                queue_count += 1
+                                
+                        except (json.JSONDecodeError, ValueError) as e:
+                            logger.warning(f"Failed to parse item from {queue_name}: {e}")
+                            continue
+                    
+                    if queue_count > 0:
+                        logger.info(f"Found {queue_count} existing issues in {queue_name}")
+                        
+                except Exception as e:
+                    logger.warning(f"Error checking {queue_name}: {e}")
+                    continue
+            
+            logger.info(f"Found {len(existing_keys)} total existing issues across all queues")
+            return existing_keys
+            
+        except Exception as e:
+            logger.error(f"Error checking existing queue items: {e}")
+            return set()
+    
+    async def push_issues_to_queue(self, issues: List[Dict[str, Any]]) -> int:
+        """
+        Push each issue to the Redis triage_queue, but only if it doesn't already exist
+        """
+        if not issues:
+            logger.info("No issues to push to queue")
+            return 0
+        
+        async with self.redis_client() as redis_conn:
+            # Get existing issue keys to avoid duplicates
+            existing_keys = await self._get_existing_issue_keys(redis_conn)
+            
+            # Extend existing_keys with issues that have jötnar labels (except jotnar_retry_needed)
+            for issue in issues:
+                issue_key = issue.get("key")
+                if issue_key:
+                    fields = issue.get("fields", {})
+                    labels = fields.get("labels", [])
+                    jotnar_labels = [label for label in labels if label.startswith('jotnar_')]
+                    
+                    # If issue has jötnar labels other than jotnar_retry_needed, mark as existing
+                    if jotnar_labels and jotnar_labels != ['jotnar_retry_needed']:
```

Comment:
from the workflow diagram, I thought if there is `jotnar_retry_needed`, despite what the rest of the labels are set, the task will be retried, am I wrong?